### PR TITLE
event: add Windows implementation

### PR DESCRIPTION
**Pull request by yaelharel**
Description:
Small changes to make the event loop work on Windows.

Risk Level:
Low

Testing:
bazel build //source/... && bazel test //test/...

Docs Changes: 
N/A

Release Notes:
N/A

Relates to issue #129

cc: @achasveachas

### CODE REVIEW

File: source/common/event/file_event_impl.cc [hunk 1]
```diff
@@ -13,6 +13,10 @@ namespace Event {
 FileEventImpl::FileEventImpl(DispatcherImpl& dispatcher, int fd, FileReadyCb cb,
                              FileTriggerType trigger, uint32_t events)
     : cb_(cb), base_(&dispatcher.base()), fd_(fd), trigger_(trigger) {
+#ifdef WIN32
+  RELEASE_ASSERT(trigger_ == FileTriggerType::Level,
```
Comment (mattklein123) [inline source/common/event/file_event_impl.cc]: Don't we generally use edge triggering throughout the code base? How does this work? Or are there follow on changes here? Also, I assume the initial goal is to just get this working vs. having good performance?

File: source/common/event/libevent_scheduler.cc [hunk 2]
```diff
@@ -15,7 +15,18 @@ TimerPtr LibeventScheduler::createTimer(const TimerCb& cb) {
...
 };
 
-void LibeventScheduler::nonBlockingLoop() { event_base_loop(libevent_.get(), EVLOOP_NONBLOCK); }
+void LibeventScheduler::nonBlockingLoop() {
+#ifdef WIN32
+  // On Windows, EVLOOP_NONBLOCK will cause the libevent event_base_loop to run forever.
+  // This is because libevent only supports level triggering on Windows, and so the write
+  // event callbacks will trigger every time through the loop. Adding EVLOOP_ONCE ensures the
+  // loop will run at most once
+  int flag = EVLOOP_NONBLOCK | EVLOOP_ONCE;
```
Comment (mattklein123) [inline source/common/event/libevent_scheduler.cc]: nit: as long as you are in here can you add const to here and below.

File: source/common/event/file_event_impl.cc (hunk 1 above)
Comment (yaelharel) [reply source/common/event/file_event_impl.cc]: Issue #4952 relates to your question.
There are follow on changes and currently we're trying to make it work first.

File: source/common/event/libevent_scheduler.cc (hunk 2 above)
Comment (yaelharel) [reply source/common/event/libevent_scheduler.cc]: Done.

**Pull request merged**